libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKSalesforceSDKUpgradeManager.m

```diff
@@ -42,18 +43,26 @@ @implementation SFSDKSalesforceSDKUpgradeManager
 + (void)upgrade {
     NSString *lastVersion = [SFSDKSalesforceSDKUpgradeManager lastVersion];
     NSString *currentVersion = [SFSDKSalesforceSDKUpgradeManager currentVersion];
-    
+
     if ([currentVersion isEqualToString:lastVersion]) {
         return;
     }
     
-    if (!lastVersion || [lastVersion doubleValue] < 9.2) {
+    if (!lastVersion || [lastVersion compare:@"9.2.0" options:NSNumericSearch] == -1) {
```

Comment:
Or is that test equivalent? "version older from 9.2" changed to "version is not 9.2". Do we want to enter the if with 9.2.1, what about future releases?

Reply:
It should be -1 (NSOrderedAscending) for older, 0 (NSOrderedSame), and 1 (NSOrderedDescending) for newer. Looking at it again I'm thinking I should use the enum name instead of the raw value. Let me know if you think there's a better way to do the comparison, too

Reply:
My bad, I forgot how compare works. I was thinking (indexOf() == -1).